src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -629,14 +635,13 @@ private Map<String, String> parseQualifiers(final String encodedString) throws M
         }
     }
 
-    @SuppressWarnings("StringSplitter")//reason: surprising behavior is okay in this case
-    private String[] parsePath(final String value, final boolean isSubpath) throws MalformedPackageURLException {
+    private static String[] parsePath(final String value, final boolean isSubpath) {
         if (value == null || value.isEmpty()) {
             return null;
         }
-        return PATH_SPLITTER.splitAsStream(value)
-                .filter(segment -> !segment.isEmpty() && !(isSubpath && (".".equals(segment) || "..".equals(segment))))
-                .map(segment -> percentDecode(segment))
+
+        return Arrays.stream(percentDecode(value).split("/"))
```

Comment:
```suggestion
        return Arrays.stream(value.split("/")).map(PackageURL::percentDecode)
```

Reply:
I think the original ordering is wrong. The change is to allow `"%2E"` or `"%2F"` in the input. This is my reasoning: If you split first and then decode second, it won't work. You have to decode first and then split. Otherwise, you'd have to split, decode, and split again, to handle the cases where the decoding process has added additional `'.'` or `'/'` into the input.

Reply:
I was trying to handle `"%2E"` and may have introduced a bug by allowing the segment to be empty. I may have confused `"%2E"` and `"%2F"`. I'll have to think about it. But, hopefully you see my logic as to why the decoding has to happen first. But, that's because the code currently combines the checking. Maybe it's clear to have two separate methods.

Reply:
I agree on the need of decoding `%2E`: dot `.` is an unreserved character, so URIs containing `.` or `%2E` are equivalent.

URIs containing `%2F` or `/` are not equivalent.  The spec says:

> the '/' used as `type`/`namespace`/`name` and `subpath` segments separator does not need to and must NOT be percent-encoded. It is unambiguous unencoded everywhere.

So if `%2F` occurs, it should not be used as separator. It will be decoded as part of a segment and the per-segment validator below will throw.

Reply:
See, this is not clear to be from the spec either. One issue is that the section talking about what it can't contain comes first, but there's no mention about it later here:

Reply:
`/` and `%2F` are not the same according to RFC 3986, since `/` is `reserved`.

There might be some security concerns in decoding `%2F` too early. Section [Backend Transcoding](https://datatracker.ietf.org/doc/html/rfc3986#section-7.3) says:

> Applications must split the URI into its components and subcomponents prior to decoding the octets, as otherwise the decoded octets might be mistaken for delimiters.

I would apply the same principle to `namespace` and `path`: it needs to be split into segments prior to decoding.

Reply:
This should be fixed in the latest commit. There should be no change to the ordering.